### PR TITLE
Fix UBSan report in arraySlice and substring

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

https://clickhouse-test-reports.s3.yandex.net/19378/2c42600cf9bd8e9962b5b5a0ebb4bf11a10887ea/fuzzer_ubsan/server.log
https://clickhouse-test-reports.s3.yandex.net/19442/ccf332ea7e459a28c6d05b90d2783aad831b91ea/fuzzer_ubsan/server.log

### CODE REVIEW

File: tests/queries/0_stateless/01664_array_slice_ubsan.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+SELECT arraySlice(groupArray(x), -9223372036854775808, NULL) AS y FROM (SELECT '6553.5', uniqState(NULL) AS x FROM numbers(3) GROUP BY number);
```
Comment (azat) [inline tests/queries/0_stateless/01664_array_slice_ubsan.sql:1]: There is also UB for `select arraySlice([],  -9223372036854775808)` (in `sliceDynamicOffsetUnbounded`)

**Comment by azat** [comment]
I have similar patch, likely I did not submit PR yet, but I noticed one more UB:

```
SELECT toIntervalMinute(9223372036854775807); -- INT64_MIN+1
SELECT toIntervalMinute(9223372036854775808); -- INT64_MIN
```

Looks like it is OK to add `NO_SANITIZE_UNDEFINED` for this function.

https://clickhouse-test-reports.s3.yandex.net/19381/95c9e57ec13e450ec6535cf723f504a2e68e3e2c/fuzzer_ubsan/report.html#fail1

**Comment by alexey-milovidov** [comment]
Could you please submit the PR nevertheless?
I will integrate the changes...

Yes, it's Ok to add NO_SANITIZE_UNDEFINED.

**Comment by azat** [comment]
>Could you please submit the PR nevertheless?

Sure - #19466

*P.S. I've removed bits that you already addressed*

**Pull request merged**